test/javascripts/components/highlighted-code-test.js

```diff
@@ -20,3 +20,22 @@ componentTest("highlighting code", {
     );
   }
 });
+
+componentTest("large code blocks are not highlighted", {
+  template: "{{highlighted-code lang='ruby' code=code}}",
+
+  beforeEach() {
+    Discourse.HighlightJSPath =
+      "assets/highlightjs/highlight-test-bundle.min.js";
+    this.set("code", "puts a\n".repeat(15000));
```

Comment:
The `"puts a\n".repeat(15000)` might be time-consuming / CPU-intensive. I think we should only do it once and put it in a `const`?